src/rqt_robot_steering/robot_steering.py

```diff
@@ -149,8 +149,14 @@ def _on_topic_changed(self, topic):
             self._publisher = rospy.Publisher(topic, Twist)
 
     def _on_stop_pressed(self):
-        self._widget.x_linear_slider.setValue(0)
-        self._widget.z_angular_slider.setValue(0)
+        # If the current value of sliders is zero directly send stop twist msg
+        if self._widget.x_linear_slider.value() is 0 and \
+                self._widget.z_angular_slider.value() is 0:
```

Comment:
Why should the stop message not be sent immediately when the sliders are at different values? That implies that it will take until the next timeout (up to 100ms atm) until the message is sent.

Reply:
According to what I can infer from the code (and also tested) When the sliders are at a different value by clicking on the stop button, their `setValue` method would be called which triggers the`valueChanged` signal (similar to cases where you move the slider). Accordingly, the `on_x/z_linear_slider_changed` slots will be executed and stop message will be sent immediately. Also, in this way we avoid from unnecessary `slider.setValue(0)` executions while the value is already 0.

Reply:
Thanks for the detailed response. Sounds good to me. Merging...